AmplifyPlugins/Analytics/AWSPinpointAnalyticsPlugin/Dependency/Pinpoint/Endpoint/EndpointClient.swift

```diff
@@ -168,7 +187,7 @@ actor EndpointClient: EndpointClientBehaviour {
         do {
             let output = try await pinpointClient.updateEndpoint(input: input)
             if let encodedData = try? archiver.encode(endpointProfile) {
-                userDefaults.save(encodedData, forKey: Constants.endpointProfileKey)
+                try keychain.set(encodedData, key: Constants.endpointProfileKey)
```

Comment:
If this operation fails, the error will propagate to the `catch` block and be rethrown, which is not desired as the `updateEndpoint` operation was actually successful on line 188. 

Either use `try?` instead or create another `catch` block that only handles this situation and does not re-throw the error.

Reply:
Updated to use `try?`